QuasarHelios2015/src/org/team1540/quasarhelios/ControlInterface.java

```diff
@@ -85,8 +77,23 @@ private static void setupRollers() {
 
         FloatInput cutoffAuto = mainTuning.getFloat("Trigger Threshold +M", 0.5f);
 
-        BooleanMixing.pumpWhen(QuasarHelios.manualControl, FloatMixing.floatIsAtLeast(Igneous.joystick2.getAxisSource(3), cutoffAuto), QuasarHelios.autoEjector);
-        BooleanMixing.pumpWhen(QuasarHelios.manualControl, FloatMixing.floatIsAtLeast(Igneous.joystick2.getAxisSource(4), cutoffAuto), QuasarHelios.autoLoader);
+        FloatInput leftTrigger = Igneous.joystick2.getAxisSource(3);
+        FloatInput rightTrigger = Igneous.joystick2.getAxisSource(4);
+
+        BooleanInputPoll rollersMode = Igneous.joystick2.getButtonChannel(5);
+
+        BooleanInput leftTriggerPress = FloatMixing.floatIsAtLeast(leftTrigger, cutoffRollers);
+        BooleanInput rightTriggerPress = FloatMixing.floatIsAtLeast(rightTrigger, cutoffRollers);
+
+        EventMixing.filterEvent(rollersMode, true, BooleanMixing.onPress(leftTriggerPress)).send(Rollers.leftPneumaticOverride.getSetTrueEvent());
+        EventMixing.filterEvent(rollersMode, true, BooleanMixing.onRelease(leftTriggerPress)).send(Rollers.leftPneumaticOverride.getSetFalseEvent());
+        EventMixing.filterEvent(rollersMode, true, BooleanMixing.onPress(rightTriggerPress)).send(Rollers.rightPneumaticOverride.getSetTrueEvent());
+        EventMixing.filterEvent(rollersMode, true, BooleanMixing.onRelease(rightTriggerPress)).send(Rollers.rightPneumaticOverride.getSetFalseEvent());
```

Comment:
Why use the press and release events when you're just conditionally pumping the value?